packages/react/src/types.ts

```diff
@@ -1,12 +1,23 @@
-import type {ReactNode, ComponentType} from 'react';
-import type {
-  RemoteComponentType,
-  PropsForRemoteComponent,
-} from '@remote-ui/core';
+import type {ReactNode, ComponentType, ReactElement} from 'react';
+import type {RemoteComponentType, RemoteFragment} from '@remote-ui/core';
+
+type PropsForRemoteComponent<T> = T extends RemoteComponentType<
+  string,
+  infer Props,
+  any
+>
+  ? {[K in keyof Props]: RemoteFragmentToReactElement<Props[K]>}
+  : never;
+
+type RemoteFragmentToReactElement<T> = T extends RemoteFragment<infer R>
+  ? ReactElement | false | RemoteFragment<R>
+  : T;
 
 export type ReactPropsFromRemoteComponentType<
```

Comment:
We probably need to have another generic argument to control whether we actually map the fragment props to React elements, something that matches the option you have for `createRemoteReactComponent`

Reply:
I think skipping this is fine for now until we have more complex use case. Currently, even with pass the options below, we only check to use if `fragmentProps` is empty or not. The actual logic still uses `isReactElement`. I found this is the most clean way at the moment. 

```
interface Options<Props> {
  fragmentProps?: (keyof Props)[];
}
```